src/gui/sharemanager.cpp

```diff
@@ -508,29 +509,24 @@ void ShareManager::fetchShares(const QString &path)
 
 void ShareManager::slotSharesFetched(const QJsonDocument &reply)
 {
-    qDebug() << reply;
-    auto tmpShares = reply.object().value("ocs").toObject().value("data").toArray();
-    const QString versionString = _account->serverVersion();
+    qCDebug(lcSharing) << reply;
+    const auto tmpShares = reply.object().value("ocs").toObject().value("data").toArray();
+    const auto versionString = _account->serverVersion();
     qCDebug(lcSharing) << versionString << "Fetched" << tmpShares.count() << "shares";
 
     QList<SharePtr> shares;
 
-    foreach (const auto &share, tmpShares) {
-        auto data = share.toObject();
-
-        auto shareType = data.value("share_type").toInt();
-
-        SharePtr newShare;
+    for (const auto &share : tmpShares) {
+        const auto data = share.toObject();
+        const auto shareType = data.value("share_type").toInt();
 
         if (shareType == Share::TypeLink) {
-            newShare = parseLinkShare(data);
-        } else if (Share::isShareTypeUserGroupEmailRoomOrRemote(static_cast <Share::ShareType>(shareType))) {
-            newShare = parseUserGroupShare(data);
+            shares.append(parseLinkShare(data));
```

Comment:
@claucambra I don't see a good reason for this refactoring, besides removing space after `static_cast` you are basically doing the same but introducing more code for review

Reply:
while fixing the issue @claucambra would it be possible to write a test for the new method you're adding?

Reply:
The main reason for this change was to eliminate the likelihood of appending a null `newShare` to `shares`, and making the code a little shorter -- it doesn't seem like we need the `newShare` here. 

As for the space I think this was auto-done by clang-format but I can re-add

Reply:
Looking into the other points now

Reply:
@claucambra No, I am fine with removing the space. I can see your point about `newShare` being not `null`, but there is an else statement so it should never be null, that said, you can just add a check `if (newShare)` prior to adding it if that makes sense